### ISSUE TITLE
Show Changes (--diff) prompt on launch ignored

### ISSUE THREAD

**Issue by CWollinger** [ansible/awx#13780]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Bug Summary

It is not possible to override the “Show Changes” (--diff) option via “Prompt on launch” for a job template. AWX uses the defined option in the job templated. A change (enable/disable) has no effect on the PromtDetail view.

### AWX version

21.13.0

### Select the relevant components

- [X] UI
- [X] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [ ] Other

### Installation method

kubernetes

### Modifications

no

### Ansible version

2.14

### Web browser

Chrome

### Steps to reproduce

1. Create a job template and enable `Prompt on launch ` for `Show Changes`
2. Launch the template and enable `Show Changes` on the prompt detail view
3. `Show Changes` / diff mode still not used on the playbook run
4. Edit template and enable `Show Changes`
5. Try to disable `Show Changes` on the prompt detail

### Expected results

Able to override the `Show Changes` via `Prompt on launch`

### Actual results

AWX will always use the option set in the template

### Additional information

![grafik](https://user-images.githubusercontent.com/11299733/228808150-7eeb74ef-88ed-4b86-82f8-da53f6d1117b.png)
![grafik](https://user-images.githubusercontent.com/11299733/228808880-270a5c67-aef6-4b9c-a102-0dc8869aea83.png)

**Comment by fosterseth** [comment]
needs investigation

**Comment by fosterseth** [comment]
@CWollinger are you able to override show changes field when posting directly to API (and not using UI)? trying to narrow down the issue

**Comment by CWollinger** [comment]
@fosterseth yes, it is possible to override via API call, if the promt on launch is set
